### ISSUE TITLE
wav2vec 1.0 cannot replicate pretrained wav2vec_large.pt model

### ISSUE THREAD

**Issue by GarlandZhang** [facebookresearch/fairseq#4038]
#### What is your question?
1. Generally: cannot replicate model `wav2vec_large.pt` so any ideas would be great
2. How long should I be training the model? Currently I have 3 gpus with 16GB ram. 
3. Are there any significant implementation changes between when wav2vec was released vs now? 
4. Does the checkpoint with the best validation loss generally have better performance than latest checkpoint (w.r.t comparing them using the same ASR model)?
5. When I calculate validation loss for `wav2vec_large.pt` it has 42 which is really high. Is this correct?

#### Code

```
python train.py /path/manifests --save-dir /save/dir --num-workers 3 --fp16 --max-update 400000 --save-interval 1 --arch wav2vec --task audio_pretraining --min-lr 1e-06 --stop-min-lr 1e-09 --optimizer adam --lr 5e-3 --lr-scheduler cosine --skip-connections-agg --residual-scale 0.5 --log-compression --warmup-updates 500 --warmup-init-lr 1e-07 --criterion wav2vec --num-negatives 10 --max-sample-size 150000 --max-tokens 1500000 --skip-invalid-size-inputs-valid-test --tensorboard-logdir /tensorboard/dir
```

#### What have you tried?
1. Tried running the same command as mentioned under `examples/wav2vec` (repasted above) and also tried running with different hyperparameters (i.e. lr scheduling, warmup updates, batch size) but can't seem to get a comparable model to `wav2vec_large.pt` when running it against same ASR model to measure performance. I trained it until it seemed to converge which usually took only up to ~10 epochs. I also used the Librispeech dataset with train-* as train.tsv and dev-* as dev.tsv
2. Ran code with earliest wav2vec 1.0 release and received better performance results but still could not replicate wav2vec_large.pt

#### What's your environment?

 - fairseq Version: main
 - PyTorch Version: 1.9.0
 - OS: Linux
 - How you installed fairseq: `pip`
 - Build command you used (if compiling from source):
 - Python version: 3.8.12
 - CUDA/cuDNN version: 11.4
 - GPU models and configuration: 3 GPUs (16gb RAM)
 - Any other relevant information:
